### DISCUSSION TITLE
Typescript: Copilot suggestions with enums do not consider my code.

### DISCUSSION THREAD

**Discussion by serkonda7** [community/community#47248] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I have (in short) the following code:
```ts
enum TokenKind {
  lcur,
  rcur
  // ...
}

function struct_decl(){
  // ...
  expect(TokenKind.lcur)
  // ...
}
```

The line `expect(TokenKind.lcur)` was suggested by copilot with incorrect enum value:
```ts
expect(TokenKind.lbrace)		
```

Even after I changed it to a existing value, copilot still suggest something invalid:
```ts
expect(TokenKind.lcurly)		
```

The existing code should be taken into context to not provide bad suggestions.

**Comment by iamcjwilliams** [comment]
@serkonda7 Thank you for your feedback, Our copilot team has been notified

**Comment by ktrout1111** [comment]
Bump.

I have the same problem. Repeated blindness to enum members, instead suggesting wrong ones that don't even exist. In fact it's not limited to enums - it will often guess names of identifiers that don't exist. 90% of the time it will insist on these non-existent names instead of real ones. It does most of the work thinking up a viable name. If it just took that extra step and compared its guess to whats already there, it would be a huge improvement.

Any progress on fixing this?

Edit: In my case they are C++ scoped enum classes, not TS.

**Reply by ayim** [reply]
@ktrout1111 which editor are you writing C++ in?
